app/javascript/components/venue/VenueEdit.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react';
+import Axios from 'axios';
+
+export default class VenueEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { venue: { address: '', description: '', people_capacity: '' } };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange = (venue) => {
```

Comment:
you can move it to some abstract class and extend your components with it. also you can put headers logic there.
*optional

Reply:
yeap, i want to try it little bit later)